app/controllers/application_controller.rb

```diff
@@ -1,3 +1,12 @@
 class ApplicationController < ActionController::Base
   include Pundit::Authorization
+
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
+
+  private
+    def user_not_authorized
+      @error = "User not authorized to perform this action"
+      flash[:error] = @error
+      redirect_to controller: "products", action: "new"
```

Comment:
Redirect sould not be specific in Application controller.
Have some generic. like

```def user_not_authorized(exception)
   policy_name = exception.policy.class.to_s.underscore

   flash[:error] = t "#{policy_name}.#{exception.query}", scope: "pundit", default: :default
   redirect_back(fallback_location: root_path)
 end```